### PR TITLE
Edited repository basic structure

### LINKED ISSUE #19: Adding some good-practice structuring elements to the Git repo.
- [V] I am running the latest version
- [V] I checked the documentation and found no answer
- [V] I checked to make sure that this issue has not already been filed

## Expected Behavior
This repo could benefit from some state-of-the-art structuring elements, such as: 
* more informative README (and documentation later on).
* pre-commit hooks to keep styling changes in check.
* setup configuration to allow for better integration with future collaborators.
* etc.

## Current Behavior
* Good basic structure, although some improvements can be beneficial.

## Acceptance Criteria

Structuring changes should in now way affects the functionality of this project, eg, any associated PR should pass all tests and keep the functionality of the project intact.

### PR DESCRIPTION
**Pull request by liorjanah**
### Desired Outcome

Ideally, this repository should be clear and easy-to-use by any potential collaborator.

### Implemented Changes

To that end, I've added/edited:
* pre-commit hooks, aimed at controlling for styling differences across different contributors.
* Setup configurations (setup,cfg, setup.py, pyproject.toml).
* Added AUTHORS.rst document.
* Added contribution guidelines (under CONTRIBUTING.rst).
* Edited README to be more informative.
* Note that all documentation files were converted to RestructuredText to better integrate them later on with Sphinx to create an [ReadTheDocs](https://readthedocs.org/) webpage.

### Connected Issue/Story

Resolves #19 

#### Changelog

- [X] The CHANGELOG has been updated

#### Documentation

- [X] Docs (e.g. `README`s) were updated in this PR
- [X] This PR does not require updating any documentation

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+* Download and Install `Vagrant`_.
+* Download and Install `Virtual Box`_.
+
+Run the project on your local machine
+======================================
+A step by step series that tell you how to get Hotails up and running:
+
+1. Clone the repository
+2. Open command prompt in Hotails directory
```
Comment (jeniaSakirko) [inline README.rst:34]: Hotails -> ResearchMate

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+3. Spin the environment using vagrant up
+4. That's it! your virtual machine is up and running
+5. Go to http://localhost:8000 in your web browser
```
Comment (jeniaSakirko) [inline README.rst:37]: Going to 8000 will bring to the API 
For theUI the user should go to 3000

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+* Download and Install `Vagrant`_.
+* Download and Install `Virtual Box`_.
+
+Run the project on your local machine
+======================================
+A step by step series that tell you how to get Hotails up and running:
+
+1. Clone the repository
+2. Open command prompt in Hotails directory
+3. Spin the environment using vagrant up
```
Comment (jeniaSakirko) [inline README.rst:35]: wrap the vagrant up to be code sample using ``
The result will be `vagrant up`

File: README.rst [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+4. That's it! your virtual machine is up and running
+5. Go to http://localhost:8000 in your web browser
+6. You can enter the virtual machine by running vagrant ssh
```
Comment (jeniaSakirko) [inline README.rst:38]: wrap vagrant ssh to be code sample

File: README.rst [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+7. When finished, run vagrant destroy -f to tear down the environment
```
Comment (jeniaSakirko) [inline README.rst:39]: same here about the code sample

**Review by jeniaSakirko — CHANGES_REQUESTED** [review]

File: README.rst (hunk 1 above)
Comment (liorjanah) [reply README.rst]: Done.

File: README.rst (hunk 2 above)
Comment (liorjanah) [reply README.rst]: Done.

File: README.rst (hunk 3 above)
Comment (liorjanah) [reply README.rst]: Done.

File: README.rst (hunk 4 above)
Comment (liorjanah) [reply README.rst]: Done.

File: README.rst (hunk 5 above)
Comment (liorjanah) [reply README.rst]: Done.

File: .gitignore [hunk 6]
```diff
@@ -157,3 +161,4 @@ node_modules/
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
+>>>>>>> 49c828f6693ecf19d778b889013d882124cc688d
```
Comment (jeniaSakirko) [inline .gitignore:164]: I think its shouldn't be here

**Review by jeniaSakirko — CHANGES_REQUESTED** [review]

**Review by jeniaSakirko — APPROVED** [review]

**Pull request merged**